cli/code_scanner.py

```diff
@@ -818,33 +818,62 @@ def _is_subpath_of_cycode_configuration_folder(filename: str) -> bool:
-                    fg='red', nl=False)
-        context.obj["soft_fail"] = True
-    elif isinstance(e, HttpUnauthorizedError):
-        click.secho('Unable to authenticate to Cycode, your token is either invalid or has expired. '
-                    'Please re-generate your token and reconfigure it by running the `cycode configure` command',
-                    fg='red', nl=False)
-        context.obj["soft_fail"] = True
-    elif isinstance(e, ZipTooLargeError):
-        click.secho('The path you attempted to scan exceeds the current maximum scanning size cap (10MB). '
-                    'Please try ignoring irrelevant paths using the ‘cycode ignore --by-path’ '
-                    'command and execute the scan again',
-                    fg='red', nl=False)
-        context.obj["soft_fail"] = True
-    elif isinstance(e, InvalidGitRepositoryError):
-        click.secho('The path you supplied does not correlate to a git repository. Should you still wish to scan '
-                    'this path, use: ‘cycode scan path <path>’',
-                    fg='red', nl=False)
-    elif isinstance(e, click.ClickException):
+
+    # TODO(MarshalX): Create global CLI errors database and move this; create error model
+    errors = {
+        CycodeError: {
+            'soft_fail': True,
+            'code': 'cycode_error',
+            'message': 'Cycode was unable to complete this scan. '
+                       'Please try again by executing the `cycode scan` command'
+        },
+        ScanAsyncError: {
+            'soft_fail': True,
+            'code': 'scan_error',
+            'message': 'Cycode was unable to complete this scan. '
+                       'Please try again by executing the `cycode scan` command'
+        },
+        HttpUnauthorizedError: {
+            'soft_fail': True,
+            'code': 'auth_error',
+            'message': 'Unable to authenticate to Cycode, your token is either invalid or has expired. '
+                       'Please re-generate your token and reconfigure it by running the `cycode configure` command'
+        },
+        ZipTooLargeError: {
+            'soft_fail': True,
+            'code': 'zip_too_large_error',
+            'message': 'The path you attempted to scan exceeds the current maximum scanning size cap (10MB). '
+                       'Please try ignoring irrelevant paths using the ‘cycode ignore --by-path’ command '
+                       'and execute the scan again'},
+        InvalidGitRepositoryError: {
+            'soft_fail': False,
+            'code': 'invalid_git_error',
+            'message': 'The path you supplied does not correlate to a git repository. '
+                       'Should you still wish to scan this path, use: ‘cycode scan path <path>’'
+        },
+    }
+
+    # TODO(MarshalX): Extend functionality of CLI printers and move this
+    def print_error(error_data) -> None:
```

Comment:
we prefer to not create "internal" functions even if their only use is here. please move it and call it `_print_error`